versioned/tiered/gc/src/main/java/com/dremio/nessie/versioned/gc/assets/IcebergAssetKeyReader.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.gc.assets;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.iceberg.DataFileCollector;
+import org.apache.iceberg.ManifestFile;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableMetadataParser;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.spark.util.SerializableConfiguration;
+
+import com.dremio.nessie.model.IcebergTable;
+import com.dremio.nessie.versioned.AssetKey;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+public class IcebergAssetKeyReader<T> implements Supplier<Stream<AssetKey>> {
+
+  private final IcebergTable table;
+  private final SerializableConfiguration hadoopConfig;
+
+  public IcebergAssetKeyReader(IcebergTable table, SerializableConfiguration configuration) {
```

Comment:
I think we should be passing a general gc configuration object which this then uses to build a specific configuration.

I guess your current thinking is that the we use the user running the job's credentials for the data retrieval as well. This feels half okay and half squishy... @laurentgo thoughts on best way to handle this?